lib/streamlit/runtime/credentials.py

```diff
@@ -102,6 +106,57 @@ def email_prompt() -> str:
 }
 
 
+def _send_email(email: str) -> None:
+    """Send the user's email to segment.io, if submitted"""
+
+    if email is None or "@" not in email:
+        LOGGER.error("That doesn't look like an email :(")
```

Comment:
nit: I think this could lead to having this error printed two times since it already gets verified here:

https://github.com/streamlit/streamlit/blob/47a1076cbc9053a754ec7042411515496c0065c4/lib/streamlit/runtime/credentials.py#L299

with an error printed here:

https://github.com/streamlit/streamlit/blob/47a1076cbc9053a754ec7042411515496c0065c4/lib/streamlit/runtime/credentials.py#L332

So, we can probably remove this error logging here.

Reply:
Good call